### PR TITLE
finish migrating ssh plugin to config system

### LINKED ISSUE #49254: Disabling host key checking via variable doesn't seem to work.
##### SUMMARY
I want to disable key checking for a small subset of the systems I manage.  I do not want to disable it via INI/Environment since that applies to all systems.  The documentation on the paramiko_ssh page seems to say that I can set variables to disable key checking.  This isn't working.  I am not entirely sure if I am reading the docs wrong, or if this is a bug.

- The https://docs.ansible.com/ansible/latest/plugins/connection/paramiko_ssh.html mentions these variables.
  - var: ansible_host_key_checking
  - var: ansible_ssh_host_key_checking
  - var: ansible_paramiko_host_key_checking

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
- paramiko_ssh
- aruba_command

##### ANSIBLE VERSION

```paste below
ansible 2.7.2
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.13 (default, Sep 26 2018, 18:42:22) [GCC 6.3.0 20170516]
```

##### CONFIGURATION

```paste below
```

##### OS / ENVIRONMENT

The system running ansible is Debian 9.x ansible is installed via pip, I am attempting to connect to an Aruba(HP) switch.

##### STEPS TO REPRODUCE

I attempt to run my playbook and I get a host key error.

ansible-playbook -i network_equipment.yml get_switch_status.yml

Inventory network_equipment.yml

```yaml
switches:
  vars:
    ansible_connection: local
    ansible_network_os: aruba
    ansible_user: manager
    ansible_ssh_pass: hunter2
    # shotgun! One of these should work right?  Right?!
    ansible_host_key_checking: False
    ansible_ssh_host_key_checking: False
    ansible_paramiko_host_key_checking: False
    host_key_checking: False
    ssh_host_key_checking: False
    paramiko_host_key_checking: False
  hosts:
    sw-core:
      ansible_host: 10.40.250.254
```

Playbook get_switch_status.yml

```yaml
---
- hosts: switches
  gather_facts: no
  tasks:

  - debug:
      var: hostvars[inventory_hostname]
      verbosity: 1

  - name: run commands to gather information
    aruba_command:
      commands:
      - show system
```

##### ACTUAL RESULTS

The `aruba_command` task fails to connect and gives an error about host key. 

Failing output
```
ansible-playbook -i network_equipment.yml get_switch_status.yml  -vvvv
ansible-playbook 2.7.2
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.13 (default, Sep 26 2018, 18:42:22) [GCC 6.3.0 20170516]
No config file found; using defaults
setting up inventory plugins
/srv/ansible/foo/network_equipment.yml did not meet host_list requirements, check plugin documentation if this is unexpected
/srv/ansible/foo/network_equipment.yml did not meet script requirements, check plugin documentation if this is unexpected
Parsed /srv/ansible/foo/network_equipment.yml inventory source with yaml plugin
Loading callback plugin default of type stdout, v2.0 from /usr/local/lib/python2.7/dist-packages/ansible/plugins/callback/default.pyc

PLAYBOOK: get_switch_status.yml ***************************************************************************************
1 plays in get_switch_status.yml

PLAY [switches] *******************************************************************************************************
META: ran handlers

TASK [debug] **********************************************************************************************************
task path: /srv/ansible/foo/get_switch_status.yml:6
ok: [sw-core] => {
    "hostvars[inventory_hostname]": {
        "ansible_check_mode": false, 
        "ansible_connection": "local", 
        "ansible_diff_mode": false, 
        "ansible_facts": {}, 
        "ansible_forks": 5, 
        "ansible_host": "10.40.250.254", 
        "ansible_host_key_checking": false, 
        "ansible_inventory_sources": [
            "/srv/ansible/foo/network_equipment.yml"
        ], 
        "ansible_network_os": "aruba", 
        "ansible_paramiko_host_key_checking": false, 
        "ansible_playbook_python": "/usr/bin/python", 
        "ansible_run_tags": [
            "all"
        ], 
        "ansible_skip_tags": [], 
        "ansible_ssh_host_key_checking": false, 
        "ansible_ssh_pass": "hunter2", 
        "ansible_user": "manager", 
        "ansible_verbosity": 4, 
        "ansible_version": {
            "full": "2.7.2", 
            "major": 2, 
            "minor": 7, 
            "revision": 2, 
            "string": "2.7.2"
        }, 
        "group_names": [
            "switches"
        ], 
        "groups": {
            "all": [
                "sw-core"
            ], 
            "switches": [
                "sw-core"
            ], 
            "ungrouped": []
        }, 
        "host_key_checking": false, 
        "inventory_dir": "/srv/ansible/foo", 
        "inventory_file": "/srv/ansible/foo/network_equipment.yml", 
        "inventory_hostname": "sw-core", 
        "inventory_hostname_short": "sw-core", 
        "omit": "__omit_place_holder__0f34100f5816f3b294f4bdaf38cc99ee6e960515", 
        "paramiko_host_key_checking": false, 
        "playbook_dir": "/srv/ansible/foo", 
        "ssh_host_key_checking": false
    }
}

TASK [run commands to gather information] *****************************************************************************
task path: /srv/ansible/foo/get_switch_status.yml:9
<10.40.250.254> using connection plugin network_cli (was local)
<10.40.250.254> starting connection from persistent connection plugin
<10.40.250.254> local domain socket does not exist, starting it
<10.40.250.254> control socket path is /root/.ansible/pc/0aa7a40423
<10.40.250.254> 
The full traceback is:
Traceback (most recent call last):
  File "/usr/local/bin/ansible-connection", line 106, in start
    self.connection._connect()
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/connection/network_cli.py", line 304, in _connect
    ssh = self.paramiko_conn._connect()
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/connection/paramiko_ssh.py", line 249, in _connect
    self.ssh = SSH_CONNECTION_CACHE[cache_key] = self._connect_uncached()
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/connection/paramiko_ssh.py", line 365, in _connect_uncached
    raise AnsibleConnectionFailure(msg)
AnsibleConnectionFailure: paramiko: The authenticity of host '10.40.250.254' can't be established.
The ssh-rsa key fingerprint is 034f0919ed2548ecddede5148a46540d.

fatal: [sw-core]: FAILED! => {
    "msg": "paramiko: The authenticity of host '10.40.250.254' can't be established.\nThe ssh-rsa key fingerprint is 034f0919ed2548ecddede5148a46540d."
}
	to retry, use: --limit @/srv/ansible/foo/get_switch_status.retry

PLAY RECAP ************************************************************************************************************
sw-core                    : ok=1    changed=0    unreachable=0    failed=1   
```

##### EXPECTED RESULTS

I expected based on what I saw in the docs that I would be able to set a host/group variable that woould permit me to disable key checking.  Like setting how the key checking in the INI or ENV works

```
ANSIBLE_HOST_KEY_CHECKING=false ansible-playbook -i network_equipment.yml get_switch_status.yml 

PLAY [switches] *******************************************************************************************************

TASK [debug] **********************************************************************************************************
skipping: [sw-core]

TASK [run commands to gather information] *****************************************************************************
ok: [sw-core]

PLAY RECAP ************************************************************************************************************
sw-core                    : ok=1    changed=0    unreachable=0    failed=0
```

### LINKED ISSUE #57220: Ansible ignores ssh configuration environment variables
##### SUMMARY
[Documentation](https://docs.ansible.com/ansible/latest/plugins/connection/ssh.html) specifies many environment variables which should be accepted by Ansible. In fact just few of them works, most not
 
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ssh

##### ANSIBLE VERSION
I tried this on 2.7 and 2.8

##### STEPS TO REPRODUCE
export ANSIBLE_SSH_COMMON_ARGS='-o ServerAliveInterval=30'
ansible-playbook -c ssh -i hosts test.yml -vvv

##### EXPECTED RESULTS
"-o ServerAliveInterval=30" is applied

##### ACTUAL RESULTS
Environment variable is ignored
```
<X.X.X.X> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="centos"' -o ConnectTimeout=10 -o ControlPath=/xxx/.ansible/cp/ba0b723c16 X.X.X.X '/bin/sh -c '"'"'echo ~centos && sleep 0'"'"''
```

### LINKED ISSUE #58184: `meta: reset_connection` ignoring variables
##### SUMMARY
meta reset_connection ignores the connection variables specified within host or group vars. Therefore connections fail with some connectors.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
`lib/ansible/modules/utilities/helper/meta.py`
`lib/ansible/plugins/connection/vmware_tools.py`

##### ANSIBLE VERSION

```paste below
ansible 2.9.0.dev0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/PycharmProjects/ansible-2/lib/ansible
  executable location = /home/user/PycharmProjects/ansible-2/bin/ansible
  python version = 3.7.3 (default, Mar 26 2019, 21:43:19) [GCC 8.2.1 20181127]
```

##### OS / ENVIRONMENT

ArchLinux

##### STEPS TO REPRODUCE

Trying to use reset_connection with e.g. vmware_tools connector fail as meta does not forward required variables, like ansible does for normal connections.

```yaml
- name: reset connection
  meta: reset_connection
```

##### EXPECTED RESULTS

successfully reconnecting to the vm (e.g. for credential switching)

##### ACTUAL RESULTS

```paste below
ERROR! Unexpected Exception, this is probably a bug: 'No setting was provided for required configuration plugin_type: connection plugin: vmware_tools setting: vmware_host '
the full traceback was:

Traceback (most recent call last):
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/plugins/__init__.py", line 58, in get_option
    option_value = C.config.get_config_value(option, plugin_type=get_plugin_class(self), plugin_name=self._load_name, variables=hostvars)
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/config/manager.py", line 381, in get_config_value
    keys=keys, variables=variables, direct=direct)
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/config/manager.py", line 456, in get_config_value_and_origin
    to_native(_get_entry(plugin_type, plugin_name, config)))
ansible.errors.AnsibleError: No setting was provided for required configuration plugin_type: connection plugin: vmware_tools setting: vmware_host 

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/PycharmProjects/ansible-2/bin/ansible-playbook", line 111, in <module>
    exit_code = cli.run()
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/cli/playbook.py", line 121, in run
    results = pbex.run()
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/executor/playbook_executor.py", line 169, in run
    result = self._tqm.run(play=play)
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/executor/task_queue_manager.py", line 239, in run
    play_return = strategy.run(iterator, play_context)
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/plugins/strategy/linear.py", line 263, in run
    results.extend(self._execute_meta(task, play_context, iterator, host))
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/plugins/strategy/__init__.py", line 1068, in _execute_meta
    connection.reset()
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/plugins/connection/vmware_tools.py", line 361, in reset
    self._connect()
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/plugins/connection/vmware_tools.py", line 344, in _connect
    self._establish_connection()
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/plugins/connection/vmware_tools.py", line 286, in _establish_connection
    "host": self.vmware_host,
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/plugins/connection/vmware_tools.py", line 246, in vmware_host
    return self.get_option("vmware_host")
  File "/home/user/PycharmProjects/ansible-2/lib/ansible/plugins/__init__.py", line 60, in get_option
    raise KeyError(to_native(e))
KeyError: 'No setting was provided for required configuration plugin_type: connection plugin: vmware_tools setting: vmware_host '
```

##### Notes
There are currently also other issues that prevent reset_connection from working with the vmware_tools connector, but they are addressed within separate issues and are out of scope for this one. This issue is about the required variables not being passed to connectors by the meta helper, this may be an issue for other connectors, too.

### LINKED ISSUE #66414: meta: reset_connection not working on Ansible 2.9.1
##### SUMMARY
I add a user to a group, execute the meta module reset_connection, check the group association again, just to find that the connection is still the same. And it's still the same ssh session!

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
Meta Module : reset_connection 

##### ANSIBLE VERSION
ansible 2.9.1                                                                                                                                                                         config file = /workspace/ansible_test_2@2/ansible/ansible.cfg                                                                                                                    configured module search path = [u'/var/jenkins_home/.ansible/plugins/modules', u'/usr/shar   e/ansible/plugins/modules']                                                                                                                                                                ansible python module location = /usr/lib/python2.7/site-packages/ansible                                                                                                           executable location = /usr/bin/ansible                                                                                                                                                   python version = 2.7.5 (default, Aug  7 2019, 00:51:29) [GCC 4.8.5 20150623 (Red Hat 4.8.5-                                                                                            39)]                                      

```

##### CONFIGURATION
<!--- Paste verbatim output from "ansible-config dump --only-changed" between quotes -->
```paste below
ANSIBLE_SSH_ARGS(/workspace/ansible_test_2@2/ansible/ansible.cfg) = -o ControlMaster=auto -o ControlPersist=30m                                                                           ANSIBLE_SSH_CONTROL_PATH(/workspace/ansible_test_2@2/ansible/ansible.cfg) = %(directory)s/%%h-%%p-%%r                                                                                     DEFAULT_HOST_LIST(/workspace/ansible_test_2@2/ansible/ansible.cfg) = [u'/environmen_data']                                                                                                DEFAULT_REMOTE_USER(/workspace/ansible_test_2@2/ansible/ansible.cfg) = USER                                                                                                               DEFAULT_ROLES_PATH(/workspace/ansible_test_2@2/ansible/ansible.cfg) = [u'/etc/ansible/roles', u'/usr/share/ansible/roles', u'/${WORKSPACE}/ansible/roles', u'/workspace/ansible_test_2@2/aHOST_KEY_CHECKING(/workspace/ansible_test_2@2/ansible/ansible.cfg) = False 
```

##### OS / ENVIRONMENT
<!--- Provide all relevant information below, e.g. target OS versions, network device firmware, etc. -->
Targed OS = Red Hat Enterprise Linux Server release 7.7 (Maipo) running a normal docker-ce installation 
Source OS = Docker Container based on a CentOS v5 image, Ansible has been installed manually and it works perfectly. 


##### STEPS TO REPRODUCE
<!--- Describe exactly how to reproduce the problem, using a minimal test-case -->
Check group association with id shell command, change group association using the normal ansible module -> Check id shell command again. 
<!--- Paste example playbooks or commands between quotes below -->
```yaml
- name: Id Check 1 
  shell: id 
  register: id_output 

- name: Give output of Id Check 1 
  debug: 
   msg: '{{ id_output.stdout }}'

- name: Modify User User group association 
  user: 
   name: '{{ ansible_user }}'
   groups: docker
   append: true
   state: present

- name: reset the SSH_CONNECTION!!!!
  meta: reset_connection 

# Optional , try a docker  command 

- name: try a docker ps command                                                                                                                                                             
  shell: docker ps                                                                                                                                                                          
  register: docker_output                                                                                                                                                                   
  ignore_errors: true                                                                                                                                                                                                                                                                                                                                                               

- name: give output of docker command                                                                                                                                                       
  debug:                                                                                                                                                                                     
   msg: '{{ docker_output.stdout  }}'

- name: Id Check 2 
  shell: id 
  register: id_output_2 

- name: Give output of Id Check 2 
  debug: 
   msg: '{{ id_output_2.stdout }}'

```

<!--- HINT: You can paste gist.github.com links for larger files -->

##### EXPECTED RESULTS
<!--- Describe what you expected to happen when running the steps above -->
The ansible user is added to the docker group, SSH Connection is resetted so that group association is updated. And the second ID command reflects the new group association.  

##### ACTUAL RESULTS
<!--- Describe what actually happened. If possible run with extra verbosity (-vvvv) -->
The session is not interrupted, so that the current session is not updated with the new Group ! Thus i cannot use a docker command.
<!--- Paste verbatim command output between quotes -->
```paste below
The -vvv run of the playbook runs with the META task    .                                                                                                                                                                              META: reset connection 
So maybe i'm missing something,  But i cannot find any indicator that it's the problem with my config file or so. 
```

### LINKED ISSUE #68341: ssh connection reset and ssh tokens in controlpath
##### SUMMARY

When investigating https://github.com/ansible-community/molecule-vagrant/issues/1, I found out that ``meta: reset_connection`` is not working due to the reset() function of the ssh.py connection plugin not being able to find the connection socket. It turns out that molecule is using ``control_path = %(directory)s/%%h-%%p-%%r``. For instance, it's translated into ``ControlPath=/home/vagrant/.ansible/cp/%h-%p-%r`` on the ssh command line.
 
The reset code does :
```
        run_reset = False
        if controlpersist and len(cp_arg) > 0:
            cp_path = cp_arg[0].split(b"=", 1)[-1]
            if os.path.exists(cp_path):
                run_reset = True
        elif controlpersist:
            run_reset = True
```
Due to the content of the ControlPath argument, it will set ``cp_path``  to ``/home/vagrant/.ansible/cp/%h-%p-%r`` and of course, the ``os.path.exists(cp_path)`` will fail, making "meta: reset_connection" useless.
fwiw, looks like this bug as been introduced by the fix for https://github.com/ansible/ansible/issues/42991

A crude workaround would be changing the test to ``if b'%' in cp_path or os.path.exists(cp_path)``. It may be better to interpolate the ssh tokens but I've no idea if it's really possible and how to do that.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ssh connection plugin

##### ANSIBLE VERSION

```
ansible --version
ansible 2.9.6
  config file = /home/rtp/devel/hupstream/ansible/molecule-vagrant-zuul/t/ansible.cfg
  configured module search path = ['/home/rtp/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/rtp/.local/lib/python3.7/site-packages/ansible
  executable location = /home/rtp/.local/bin/ansible
  python version = 3.7.3 (default, Dec 20 2019, 18:57:59) [GCC 8.3.0]
```

##### CONFIGURATION
```
[defaults]
ansible_managed = Ansible managed: Do NOT edit this file manually!
display_failed_stderr = True
forks = 50
retry_files_enabled = False
host_key_checking = False
nocows = 1
interpreter_python = auto
[ssh_connection]
scp_if_ssh = True
control_path = %(directory)s/%%h-%%p-%%r
pipelining = True

```

##### OS / ENVIRONMENT
Debian stable

##### STEPS TO REPRODUCE
Since I was testing that for molecule-vagrant, I've a Vagrantfile and a converge.yml file:

Vagrantfile:
```
Vagrant.configure("2") do |c|
  c.vm.define 'test2' do |v|
    v.vm.hostname = 'test2'
    v.vm.box = 'debian/buster64'
  end
  c.vm.provision :ansible do |ansible|
    ansible.playbook = "converge.yml"
  end
end
```
converge.yml
```
---
- name: Converge
  hosts: all
  gather_facts: false
  tasks:
    - name: Create test group
      group:
        name: testgroup
      become: true
    - name: Add vagrant user to test group
      user:
        name: vagrant
        groups: testgroup
        append: yes
      become: true
    - name: reset connection
      meta: reset_connection
    - name: Get vagrant user info
      command: id -nG
      register: user_grps
    - name: Print user_grps
      debug:
        var: user_grps
    - name: Check user in vagrant group
      assert:
        that:
          - "'testgroup' in user_grps.stdout.split(' ')"
```
ansible.cfg
```
[ssh_connection]
control_path = %(directory)s/%%h-%%p-%%r
```

##### EXPECTED RESULTS
The run of the ``converge.yml`` should work

##### ACTUAL RESULTS
```
TASK [Check user in vagrant group] *********************************************
fatal: [test2]: FAILED! => {
    "assertion": "'testgroup' in user_grps.stdout.split(' ')",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}
```

### LINKED ISSUE #72739: Ansible ignores all ssh_connection vars from ansible.cfg
##### SUMMARY

Ansible ignores all and any INI values from `[ssh_connection]` configuration.

Related: https://github.com/ansible/ansible/issues/57220

Ansible [ssh plugin](https://docs.ansible.com/ansible/2.9/plugins/connection/ssh.html#ssh-connection) clearly documents `ssh_common_args ` and a bunch of other args as being configurable in `ssh_connection` INI section, but they are ignored, at least when placed in ansible.cfg in current directory.

##### ISSUE TYPE

- Bug Report

##### COMPONENT NAME

ssh

##### ANSIBLE VERSION

```
ansible 2.9.15
  config file = /home/ec2-user/ansible.cfg
  configured module search path = [u'/home/ec2-user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ec2-user/.local/lib/python2.7/site-packages/ansible
  executable location = /home/ec2-user/.local/bin/ansible
  python version = 2.7.5 (default, Aug 13 2020, 02:51:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

##### CONFIGURATION

_Note_: this is incorrect as it excludes `ssh_connection` entirely.

```paste below
DEFAULT_ASK_PASS(/home/ec2-user/ansible.cfg) = True
DEFAULT_BECOME_ASK_PASS(/home/ec2-user/ansible.cfg) = True
DEFAULT_LOG_PATH(/home/ec2-user/ansible.cfg) = /home/ec2-user/ansible.log
DEFAULT_STDOUT_CALLBACK(/home/ec2-user/ansible.cfg) = full_skip
HOST_KEY_CHECKING(/home/ec2-user/ansible.cfg) = False

```

##### OS / ENVIRONMENT

RHEL 7.9 control + managed node.

##### STEPS TO REPRODUCE

Edit ansible.cfg:

```ini
[ssh_connection]
ssh_common_args = -o UserKnownHostsFile=/dev/null
```

Create any basic one-task playbook.

```bash
ansible-playbook -i hosts.yaml -kK --private-key ~/.ssh/<keyname>.pem -vvvv <playbook>.yaml
```

##### EXPECTED RESULTS

`ssh_common_args` from `ansible.cfg` should be passed on to `ssh` command.

##### ACTUAL RESULTS

It's not.

```
Using /home/ec2-user/ansible.cfg as config file
...
SSH: EXEC sshpass -d10 ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="/home/ec2-user/.ssh/<xxx>.pem"' -o ConnectTimeout=10 -o ControlPath=/home/ec2-user/.ansible/cp/xxxxxxx zz.zz.zz.zz '/bin/sh -c ...
```

### PR DESCRIPTION
**Pull request by bcoca**
fixes #72739
  fixes #57220
  fixes #49254
  fixes #68341
  fixes #66414
  fixes #58184

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

connection/ssh

### CODE REVIEW

**Comment by mkrizek** [comment]
!waffling core_review

**Comment by mkrizek** [comment]
!waffling core_review

File: lib/ansible/plugins/connection/ssh.py [hunk 1]
```diff
@@ -1251,23 +1297,25 @@ def fetch_file(self, in_path, out_path):
...
+        # If we have a persistent ssh connection (ControlPersist), we can ask it to stop listening.
+        # only run the reset if the ControlPath already exists or if it isn't configured and ControlPersist is set
+        # 'check' will determine this.
+        cmd = self._build_command(self.get_option('ssh_executable'), 'ssh', '-O', 'check', self.host)
+        display.vvv(u'sending connection check: %s' % to_text(cmd))
+        p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = p.communicate()
+        status_code = p.wait()
+        if status_code != 0:
+            display.vvv(u"No connection ot reset: %s" % to_text(stderr))
```
Comment (Shrews) [inline lib/ansible/plugins/connection/ssh.py]:
```suggestion
            display.vvv(u"No connection ot reset: %s" % to_text(stderr))
```
```suggestion
            display.vvv(u"No connection to reset: %s" % to_text(stderr))
```

File: lib/ansible/plugins/connection/ssh.py [hunk 2]
```diff
@@ -461,11 +511,12 @@ class Connection(ConnectionBase):
     def __init__(self, *args, **kwargs):
         super(Connection, self).__init__(*args, **kwargs)
 
+        # TODO: all should come from get_option(), but not might be set at this point yet
         self.host = self._play_context.remote_addr
         self.port = self._play_context.port
         self.user = self._play_context.remote_user
```
Comment (Shrews) [inline lib/ansible/plugins/connection/ssh.py:517]: self.user is being set below now, so is this needed?

Comment (bcoca) [reply lib/ansible/plugins/connection/ssh.py:517]: i tried removing before and failed, but willing to give it a go once i migrate remote_addr completely

**Pull request merged**